### PR TITLE
Add more info lock

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

https://github.com/home-assistant/frontend/assets/5878303/e2ee7fc4-b26f-420c-89cf-ca825d38027f

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Can `state-summary/state-card-lock.js` be removed with this PR?

**Comment by piitaya** [comment]
> Can `state-summary/state-card-lock.js` be removed with this PR?

No, it's used for group more info dialog

File: src/dialogs/more-info/controls/more-info-lock.ts [hunk 1]
```diff
@@ -1,43 +1,156 @@
...
+        return;
+      }
+      code = response;
+    }
+
+    this.hass.callService("lock", service, {
+      entity_id: this.stateObj!.entity_id,
+      code,
+    });
+  }
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-lock.ts:60]: We have this code twice (here and `ha-more-info-lock-toggle`), can we share it?

**Review by bramkragten — APPROVED** [review]

File: src/dialogs/more-info/controls/more-info-lock.ts (hunk 1 above)
Comment (piitaya) [reply src/dialogs/more-info/controls/more-info-lock.ts:60]: We have the same for alarm_control_panel. I will do a another PR to update that

**Pull request merged**